.github/workflows/build.yml

```diff
@@ -237,6 +241,131 @@ jobs:
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Get long name
+        id: master
+        run: |
+          last_version=`git tag --list --sort=-v:refname "v*"| head -n 1`
+          n_commit=`git log $last_version..HEAD --oneline | wc -l`
+          current_sha1=`git rev-parse HEAD | cut -c 1-8`
+          echo "::set-output name=human_name::$last_version-$n_commit-$current_sha1"
+
+      - name: Check existence of nightly tag
+        id: nightly_tag
+        run: |
+          echo "::set-output name=exists::$(git tag --list | grep nightly | wc -l)"
+
+      - name: Get nightly sha1 from tag
+        if: steps.nightly_tag.outputs.exists == '1'
+        id: nightly
+        run: |
+          echo "::set-output name=sha::$(git rev-parse nightly)"
+
+      - name: Create nightly tag
+        if: steps.nightly_tag.outputs.exists == '0'
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.gh_access_token }}
+          script: |
+            await github.rest.git.createRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'refs/tags/nightly',
+              sha: context.sha,
+              force: true
+            })
+
+      - name: Create release
+        if: steps.nightly_tag.outputs.exists == '0'
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.gh_access_token }}
+          script: |
+            await github.rest.repos.createRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: 'nightly',
+              body: 'These are builds generated every night using a crontab'
+            })
+
+      - name: Update Reference and Release
+        if: (steps.nightly_tag.outputs.exists == '1') &&(github.sha != steps.nightly.outputs.sha)
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.gh_access_token }}
+          script: |
+
+            await github.rest.git.updateRef({
+              owner: context.repo.owner,
+              repo:  context.repo.repo,
+              ref:   'tags/nightly',
+              sha:   context.sha,
+              force: true,
+            });
+
+            const release = await github.rest.repos.getReleaseByTag({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag: 'nightly',
+            });
+
+            await github.rest.repos.updateRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: 'nightly',
+              release_id: release.data.id,
+              target_commitish: context.sha,
+            });
+
+      - name: Continuing only if master and nightly are different
+        if: github.sha != steps.nightly.outputs.sha
+        id: continue
+        run: |
+          echo "::set-output name=value::1"
+
```

Comment:
What is the purpose of this step? Maybe you wanted use it as a condition for each step below? But for now, they are just using the same condition ̀github.sha != steps.nightly.outputs.sha` so it looks like it is not used anywhere.